src/reflection.ts

```diff
@@ -247,11 +247,19 @@ export class Parser {
     const lambdas: Record<string, any> = {};
     const schema = this.getType(typeIndex);
     const numFields = schema.fieldsLength();
-    for (let ii = 0; ii < numFields; ++ii) {
-      const field = schema.fields(ii);
+
+    // Sort fields by ID so the resulting object is built with fields in this order
```

Comment:
Can you expand on why this matters in the comment?

Reply:
Sorry, I guess my question was "why do you care"? I'm generally happy to be making things more predictable, but if this were causing problems in the future and I was debating reverting it, I wouldn't really have enough information to figure out why we had changed it in the first place.

Reply:
I guess it’s hard to come up with an airtight argument for it but I guess my statement would be that matching the .fbs order probably results in the “least surprise” for users since the objects will come out matching what they wrote in the schema. Certainly if the fields are getting reordered, alphabetical order seems a bit arbitrary, and if we’re picking an arbitrary order we might as well pick `id`.

(As publishers of schemas ourselves, Foxglove also has an interest in making our app work in a way that surprises people as little as possible, and we’ve had some reports of surprise from the alphabetical ordering.)

Not sure how best to collapse this into a code comment but let me know your thoughts 😄

Reply:
Honestly having this in the comments here may be sufficient.

For my own edification, does this actually affect the return value of the `Record` or does this just affect what order errors will be presented if present?

Reply:
Yes, objects in JS retain their key order (with some caveats): https://stackoverflow.com/a/5525820/23649

The unit test with Object.keys() covers this (it will fail if you remove the sort())